method.rb

```diff
@@ -0,0 +1,181 @@
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    i = 0
+    arr = to_a
+    return to_enum(:my_each) unless block_given?
+
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = to_a
+    return to_enum(:my_each_with_index) unless block_given?
+
+    while i < arr.size
+      yield arr[i], i
+      i += 1
+    end
+
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+
+    new_arr
+  end
+
+  def my_all?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = false unless yield num
+      end
+    else
+      my_each do |num|
+        bool = !num
+        stat = !bool
+      end
+    end
+    stat
+  end
+
+  def my_any?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = false
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+
+        my_each do |num|
+          stat = num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = true if yield num
+      end
+    else
+      my_each do |num|
+        bool = !num
+        stat = !bool
+      end
+    end
+    stat
+  end
+
+  def my_none?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = !num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = !num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = false if yield num
+      end
+    else
+      my_each do |num|
+        stat = !num
+      end
+    end
+    stat
+  end
+
+  def my_count(counter = nil)
+    p_count = 0
+    return size if counter.nil? && !block_given?
+
+    if block_given?
+      my_each do |num|
+        p_count += 1 if yield num
+      end
+
+    end
+    my_each do |num|
+      p_count += 1 if num == counter
+    end
+    p_count
+  end
+
+  def my_inject(acc = 0, sign = nil)
```

Comment:
- [ ] my_inject should also be able to take a block.
Make sure it works with for example an array or strings and a block that adds only the strings longer than 3 characters.